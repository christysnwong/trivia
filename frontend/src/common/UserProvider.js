import React, {useEffect, useState} from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  stats: {
    level: 14,
    title: "Pro",
    "quizzes completed": 34,
    points: 3440,
    remainingPts: 60,
    levelPts: 300,
  },
};

const UserProvider = ({
  children,
  currUser = demoUser
}) => {
  
  const [currentUser, setCurrUser] = useState();

  return (
    // <UserContext.Provider value={{ currUser }}>
    <UserContext.Provider value={{ currUser, setCurrUser }}>
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
