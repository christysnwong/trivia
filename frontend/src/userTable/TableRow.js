import { useEffect, useState, useContext } from "react";
import UserContext from "../common/UserContext";

const TableRow = ({ id, entry, leaderboard }) => {
  const [isUser, setIsUser] = useState(false);
  const { currUser } = useContext(UserContext);

  // check if user on record is the same as logged in user
  useEffect(() => {
    if (currUser) {
      if (entry.username === currUser.username) {
        setIsUser(true);
      }
    }
  }, [entry, currUser]);

  return (
    <tr>
      <th scope="row">
        {id + 1}
        {leaderboard && isUser && <i className="fa-solid fa-star me-2"></i>}
      </th>

      {/* check to see if the table is used for leaderboard and if the record belongs to user, set red font */}
      {(!leaderboard || !isUser) &&
        Object.values(entry).map((field, idx) => (
          <td key={idx} className="text-capitalize">
            {field}
          </td>
        ))}

      {leaderboard &&
        isUser &&
        Object.values(entry).map((field, idx) => (
          <td key={idx} className="text-capitalize text-danger fw-bold">
            {field}
          </td>
        ))}
    </tr>
  );
};

export default TableRow;
