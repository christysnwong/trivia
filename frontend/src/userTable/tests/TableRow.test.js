import React from "react";
import { render } from "@testing-library/react";
import TableRow from "../TableRow";
import { MemoryRouter } from "react-router-dom";
import UserProvider from "../../common/UserProvider";

const entry = {
  category: "Entertainment: Books",
  difficulty: "easy",
  score: 9,
  points: 70,
  date: "Sep 8th 2022, 1:20pm",
};


it("renders without crashing", function () {
  render(
    <UserProvider>
      <TableRow id={1} entry={entry} />
    </UserProvider>,
    { container: document.createElement("tbody") }
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <UserProvider>
      <TableRow id={1} entry={entry} />
    </UserProvider>,
    { container: document.createElement("tbody") }
  );
  expect(asFragment()).toMatchSnapshot();
});

