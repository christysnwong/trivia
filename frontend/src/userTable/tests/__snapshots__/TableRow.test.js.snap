// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  2Entertainment: Bookseasy970Sep 8th 2022, 1:20pm
</DocumentFragment>
`;
